### PR TITLE
[demo] add imageprovider

### PR DESCRIPTION
**Pull request by klucsik**
Update demo chart to include https://github.com/open-telemetry/opentelemetry-demo/pull/1462

### CODE REVIEW

**Comment by puckpuck** [comment]
Because this is adding a new service that has not yet been released, we need to wait for the next demo release before merging this.

File: charts/opentelemetry-demo/values.yaml [hunk 1]
```diff
@@ -385,6 +389,27 @@ components:
...
+      - name: OTEL_COLLECTOR_HOST
+        value: $(OTEL_COLLECTOR_NAME)
+    resources:
+      limits:
+        memory: 50Mi
+    securityContext:
+      runAsUser: 101  # nginx
+      runAsGroup: 101
+      runAsNonRoot: true
+
```
Comment (graphaelli) [inline charts/opentelemetry-demo/values.yaml:412]: is this working for you?  it is triggering `/bin/sh: 1: cannot create /etc/nginx/nginx.conf: Permission denied` here

File: charts/opentelemetry-demo/values.yaml [hunk 2]
```diff
@@ -371,6 +371,10 @@ components:
         value: "16686"
       - name: JAEGER_SERVICE_HOST
         value: '{{ include "otel-demo.name" . }}-jaeger-query'
+      - name: IMAGE_PROVIDER_PORT
+        value: "8081"
+      - name: IMAGE_PROVIDER_HOST
+        value: '{{ include "otel-demo.name" . }}-IMAGE_PROVIDER_PORT'
```
Comment (graphaelli) [inline charts/opentelemetry-demo/values.yaml:377]:
```suggestion
        value: '{{ include "otel-demo.name" . }}-imageprovider'
```

**Comment by zeitlinger** [comment]
IMAGE_PROVIDER_PORT is already required in https://github.com/grafana/opentelemetry-demo/blob/06023dd91e02d2da7818137b1e09eadd1d86228b/src/frontendproxy/envoy.tmpl.yaml#L109-L110 - so  helm chart 0.30.2 is broken

**Comment by julianocosta89** [comment]
> IMAGE_PROVIDER_PORT is already required in https://github.com/grafana/opentelemetry-demo/blob/06023dd91e02d2da7818137b1e09eadd1d86228b/src/frontendproxy/envoy.tmpl.yaml#L109-L110 - so helm chart 0.30.2 is broken

@puckpuck it seems that we already have to cut a new release.

**Comment by julianocosta89** [comment]
> IMAGE_PROVIDER_PORT is already required in https://github.com/grafana/opentelemetry-demo/blob/06023dd91e02d2da7818137b1e09eadd1d86228b/src/frontendproxy/envoy.tmpl.yaml#L109-L110 - so helm chart 0.30.2 is broken

Actually not @zeitlinger.
The Helm Chart has the image versions pinned to the last release, and `imageProvider` was not available on `1.9.0`.

**Comment by zeitlinger** [comment]
> Actually not @zeitlinger. The Helm Chart has the image versions pinned to the last release, and `imageProvider` was not available on `1.9.0`.

I'm using a fork of the otel demo - and did a rebase before I encountered the problem. Maybe that was the root cause.

**Comment by TylerHelmuth** [comment]
Has the demo had a release yet with the new service?

**Comment by graphaelli** [comment]
No afaict, 1.9.0 was released 2 days before https://github.com/open-telemetry/opentelemetry-demo/pull/1462 was merged

**Comment by julianocosta89** [comment]
> No afaict, 1.9.0 was released 2 days before https://github.com/open-telemetry/opentelemetry-demo/pull/1462 was merged

Correct, imageprovider was not released yet.

**Comment by puckpuck** [comment]
@klucsik I needed to get this running for my team's demo efforts. I figured out why the nginx pod wasn't starting (securityContext) and updated the frontendproxy config. I pushed the changes to your branch.

I will push to get a demo 1.10.x release done, so we can merge this.

**Comment by puckpuck** [comment]
With #1220 getting merged, this is no longer needed.

**Pull request closed without merging**